### DISCUSSION TITLE
Docs on Implementing PineAPPL Interface

### DISCUSSION THREAD

**Discussion by alecandido** [NNPDF/pineappl#155] [General]
We should provide documentation about:
- what PineAPPL is doing to get grids from a MC (abstract level, brief since it is partly present in the paper)
- explicit requirements for a MC to fill a grid
- practical instructions on how to fill call the PineAPPL library functions

**Comment by cschwan** [comment]
I fully agree, but I think this is a duplicate of #141. Shall we move the discussion there?

**Comment by alecandido** [comment]
> I fully agree, but I think this is a duplicate of https://github.com/N3PDF/pineappl/issues/141. Shall we move the discussion there?

Oh, you're right, I'm sorry...

Just to give you some context, and then we close:
- in particular, after the talk of Stefano Camarda, I briefly discussed with Giancarlo: most likely there will be a valid reason why DYTurbo (N3LO + N3LL resummed pt) won't be able to fill a PineAPPL grid; but in case I'd like to know, because otherwise might be very useful: they were able to fit very precise the alpha_s value from the Z pt distribution, but they had to do "an approximate PDF fit" (i.e. add some parameters to give PDFs some freedom) with xFitter -> if we can fill a grid we can try one of our frameworks for simultaneous PDF and alpha_s fits, that might be much more constrained than what is right now (I wonder...)

**Reply by cschwan** [reply]
Did you ask them whether they'd be interested in an interface to PineAPPL? I think having that would be interesting, but I think I didn't quite get your point.

**Reply by alecandido** [reply]
The idea is that:
- they selected a very clean observable for alphas determination: we know that in the NNPDF there are some alphas ambiguity, so if it's possible to include data with a cleaner handle on alphas, maybe it can help to have a better determination within the global fit
- they had to allow at least a bit of freedom on the PDF side; so if we can do an actual, so much the better

**Reply by alecandido** [reply]
I actually asked if it were possible according to them, *and implicitly* if they were interested in.
Looks like they were not too familiar with the concept of interpolation grids, but they could be interested. I have yet no answer to the question "is it possible?", if you were present maybe we would have had.

**Reply by cschwan** [reply]
I suppose doing a similar exercise would be possible. What would be new and uncertain is how to interface a resummed calculation with PineAPPL. As far as I know resummed calculations use an exponentiated series in alphas which isn't supported in PineAPPL right now. But we could of course add it.

**Reply by alecandido** [reply]
If I find I will try to discuss more. Otherwise I'll try to get in touch with him in Milan.

The resummed part would be expanded "somehow" in $\alpha_s$, but it will include resummed towers $g(\alpha_s L)$. So the idea is that partially is not expanded at all, so \alpha_s is not only in the coefficient.

At this point, I wonder how can you put this into an interpolation grid...

The other option is to give up on being "differential" on order (store orders separately), and just generate some grids for a few values of $\alpha_s$, and then interpolate the grids results.
This is what they are doing with FkTables in SIMUnet: since FkTables have $\alpha_s$ baked in because of evolution (i.e. EKO), then they just evaluate for a few values. And that's the exact same thing, since EKO is doing resummation as well (collinear, while DYTurbo is resumming $q_T$, i.e. $L = log(q_T^2 / M^2)$.

**Reply by cschwan** [reply]
If you have the results in the form

$$\exp \left( \sum_k \alpha_\mathrm{s}^k F_k \right)$$

then you expand in terms of LL, NLL, NNLL, etc. using products:

$$\prod_k \exp \left( \alpha_\mathrm{s}^k F_k \right)$$

from which you can store $F_k $ and use $ \alpha_\mathrm{s}$ from the PDF/alphas provider (LHAPDF/EKO).

**Reply by cschwan** [reply]
But that's probably too naive a view, we have to try that with a simple calculation.

**Reply by alecandido** [reply]
Here there is the definition and references:
https://arxiv.org/pdf/2103.04974.pdf

Eq. (7) writes down the resummation bit
